webapp-django/crashstats/crashstats/tests/test_views.py

```diff
@@ -2659,6 +2659,59 @@ def test_crontabber_state(self):
         response = self.client.get(url)
         eq_(response.status_code, 200)
 
+    @mock.patch('requests.get')
+    def test_crontabber_status_ok(self, rget):
+
+        def mocked_get(url, **options):
+            assert '/crontabber_state/' in url
+            return Response({
+                'state': {
+                    'job1': {
+                        'error_count': 0,
+                        'depends_on': []
+                    }
+                }
+            })
+
+        rget.side_effect = mocked_get
+
+        url = reverse('crashstats:crontabber_status')
+        response = self.client.get(url)
+        eq_(response.status_code, 200)
+        eq_(json.loads(response.content), {'status': 'OK'})
+
+    @mock.patch('requests.get')
+    def test_crontabber_status_trouble(self, rget):
+
+        def mocked_get(url, **options):
+            assert '/crontabber_state/' in url
+            return Response({
+                'state': {
+                    'job1': {
+                        'error_count': 1,
+                        'depends_on': [],
+                    },
+                    'job2': {
+                        'error_count': 0,
+                        'depends_on': ['job1'],
+                    },
+                    'job3': {
+                        'error_count': 0,
+                        'depends_on': ['job2'],
+                    }
```

Comment:
I think this would deserve to test even more things. I would add at least one non-failing test and a few more levels of dependency of failing jobs, to test that ``while`` loop in the code.